src/components/stepper/stepper/stepper.ts

```diff
@@ -0,0 +1,321 @@
+import {
+  type CSSResultGroup,
+  html,
+  LitElement,
+  type TemplateResult,
+  type PropertyValues,
+} from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+import { getNextElementIndex, isArrowKeyPressed } from '../../core/a11y.js';
+import { SbbConnectedAbortController } from '../../core/controllers.js';
+import { breakpoints, isBreakpoint } from '../../core/dom.js';
+import type { SbbHorizontalFrom, SbbOrientation } from '../../core/interfaces.js';
+import type { SbbStepElement, SbbStepValidateEventDetails } from '../step.js';
+
+import style from './stepper.scss?lit&inline';
+
+const DEBOUNCE_TIME = 150;
+
+/**
+ * Provides a structured, step-by-step workflow for user interactions.
+ *
+ * @slot step-label - Use this slot to provide an `sbb-step-label`.
+ * @slot step - Use this slot to provide an `sbb-step`.
+ */
+@customElement('sbb-stepper')
+export class SbbStepperElement extends LitElement {
+  public static override styles: CSSResultGroup = style;
+
+  /**
+   * If set to true, only the current and previous labels can be clicked and selected.
+   */
+  @property({ type: Boolean }) public linear = false;
+
+  /**
+   * Overrides the behaviour of `orientation` property.
+   */
+  @property({ attribute: 'horizontal-from', reflect: true })
+  public get horizontalFrom(): SbbHorizontalFrom | undefined {
+    return this._horizontalFrom;
+  }
+  public set horizontalFrom(value: SbbHorizontalFrom) {
+    this._horizontalFrom = breakpoints.includes(value) ? value : undefined;
+    if (this._horizontalFrom && this._loaded) {
+      this._checkOrientation();
+    }
+  }
+  private _horizontalFrom?: SbbHorizontalFrom | undefined;
+
+  /**
+   * Steps orientation, either horizontal or vertical.
+   */
+  @property({ reflect: true })
+  public orientation: SbbOrientation = 'horizontal';
+
+  /**
+   * The currently selected step.
+   */
+  @property({ attribute: false })
+  public set selected(step: SbbStepElement) {
+    if (this._loaded) {
+      this._select(step);
+    }
+  }
+  public get selected(): SbbStepElement | undefined {
+    return this.querySelector<SbbStepElement>('sbb-step[data-selected]') ?? undefined;
+  }
+
+  /**
+   * The currently selected step index.
+   */
+  @property({ attribute: 'selected-index', type: Number })
+  public set selectedIndex(index: number) {
+    if (this._loaded) {
+      this._select(this.steps[index]);
+    }
+  }
+  public get selectedIndex(): number | undefined {
+    return this.selected ? this.steps.indexOf(this.selected) : undefined;
+  }
+
+  /**
+   * The steps of the stepper.
+   */
+  public get steps(): SbbStepElement[] {
+    return Array.from(this.querySelectorAll('sbb-step'));
```

Comment:
steps gets called on SSR, where querySelectorAll is not defined
```suggestion
    return Array.from(this.querySelectorAll?.('sbb-step') ?? []);
```